Sources/Workspace/Workspace.swift

```diff
@@ -341,19 +341,10 @@ public class Workspace {
             throw WorkspaceOperationError.dependencyAlreadyInEditMode
         }
 
-        let destination: AbsolutePath
-        let state: ManagedDependency.State
-
         // If a path is provided then we make this dependency unmanaged (Top of
         // the tree). Otherwise, it is an edited dependency inside editables
         // directory.
-        if let path = path {
-            destination = path
-            state = .unmanaged(path: destination)
-        } else {
-            destination = editablesPath.appending(component: packageName)
-            state = .edited
-        }
+        let destination = self.path(for: dependency, state: .edited(path: path))
```

Comment:
Do you have a better name for this method? maybe computePath, formPath, dependencyPath etc